app/components/polls/statements/Cards.tsx

```diff
@@ -93,24 +108,25 @@ const Cards = ({
         }px`,
       }}
     >
-      {statementsToDisplay.map((statement, index) => (
-        <Card
-          key={statement.id}
-          statement={statement}
-          statementOptions={statementOptions[statement.id] ?? []}
-          index={index}
-          cardCount={statementsToDisplay.length}
-          onStatementHide={() =>
-            setStatementsToHide((hiddenStatements) => [
-              ...hiddenStatements,
-              statement.id,
-            ])
-          }
-          height={
-            index === statementsToDisplay.length - 1 ? undefined : cardHeight
-          }
-        />
-      ))}
+      {!!statementSorting?.length && // check if sorted before showing to avoid flickering
```

Comment:
I forgot, you also don't need this check anymore when you set statementSorting on the first render